src/products/service/stock.service.ts

```diff
@@ -0,0 +1,33 @@
+import { BadRequestException, Injectable, NotFoundException } from "@nestjs/common";
+import { StockRepository } from "../repository/stock.repository";
+import { OrderItem } from "src/orders/dto/create-order.dto";
+import { Stock } from "../domain/stock.entity";
+
+export interface StockItem{
+    productId: string;
+    quantity: number;
+}
+@Injectable()
+export class StockService{
+
+    constructor(private readonly stockRepository: StockRepository){}
+
+    async get(orderItems: OrderItem[]): Promise<Stock[]>{
+        let stockQueries: Promise<Stock>[];
+        try {
+            stockQueries = orderItems.map(orderItem => this.stockRepository.get(orderItem.product.id, orderItem.location.id));
```

Comment:
**issue:** iterating with multiple `get` statements transform this at database layer into multiple select statements. This in inefficient and may lead to performance issues later down the line. The correct you to do this is to have a getMany query which gets a list of ids. This is transformed into a single select statement with a `where id in [array_of_ids]` clause

Reply:
Fixed using:

![image](https://github.com/user-attachments/assets/7518414f-e379-4847-810a-4a9ee266f1c4)